### DISCUSSION TITLE
[Guide] Mounting external NFS volume from QNAP Photos (external)

### DISCUSSION THREAD

**Discussion by trumpetx** [immich-app/immich#10887] [Community Guides]
I'm writing this so I can search up this answer in 2 years when I need the information again.  If this helps anyone else, you're welcome!  I would also love to hear any suggestions for improving this setup.

This guide is NOT about running the QNAP Container Station.  This is for mounting the //NAS/Photots to your Immich server (docker) which is a separate machine.  Some of the info here is probably applicable to the other situation though. 

**Make sure permissions are set up correctly on the QNAP NAS (Privileges > Shared Folders)**
![Screenshot 2024-07-05 at 08-44-23 QNAP-NAS](https://github.com/immich-app/immich/assets/115343/51ce2ded-6946-4fd3-9404-6e9272cfa691)

1. Read only (we don't want Immich to accidentally mess something up
2. Squash All Users is the magic option which is going to let non-root users on the remote system view the files
3. Only open it up to your local network (mine, in this case is 192.168.1.x represented by 192.168.1.0/24)

**Mount the NFS drive on the HOST machine where the docker container will be started**

1. Create the folder
```
sudo mkdir -p /mnt/photos
```
2. Make sure you have the nfs utils installed:
Ubuntu 24.04:
```
sudo apt install nfs-common
```
(note: this is different on other distros, you're looking for whatever package gets you `mount` for nfs)

4. Test the mount
```
# 192.168.1.2 is my NAS ip address
sudo mount -t nfs -O "auto,nofail,noatime,nolock,intr,tcp,actimeo=1800" 192.168.1.2:Photos /mnt/photos
ls -ll /mnt/photos
# you should see a listing of directories/photos now
```
5. Make the mount permanent
```
sudo vim /etc/fstab
# Add the following line
192.168.1.2:/Photos   /mnt/photos     nfs     _netdev,x-systemd.automount,auto,nofail,noatime,nolock,intr,tcp,actimeo=1800    0       0
```
In my case, after some troubleshooting with network, I had to prepend `_netdev,x-systemd.automount` to the previously tested manual mount.  This was on a Ubuntu 24.04 system, YMMV.
6. Reboot and validate that your mount is still there as a non-root user.

**Add the External Library**

Do the steps here: https://immich.app/docs/guides/external-library.  Summary of steps:

Added to .env:
```
# Mounted NFS drives from QNAP
EXTERNAL_PATH_PHOTOS=/mnt/photos
```

Added to docker-compose.yml:
```
  immich-server:
    ...
    volumes:
      ...
      - ${EXTERNAL_PATH_PHOTOS}:/mnt/photos:ro
```

**Configure the External Library**
1. Add an import path to our mounted/remounted NFS drive: `/mnt/photos`
2. Add the following exclusions under "Scan Settings" to skip over QNAP "stuff":
  - `/mnt/photos/**/@Recycle`
  - `/mnt/photos/**/.@__thumb`

**Comment by CampRyenWater** [comment]
Thank you for taking the time to write this article. It has been helpful as I try to get an External Library working in IMMICH.

**Comment by jrobiii** [comment]
Just what I was looking for. I'm working with a Synology, but I bet it's very similar.  Thanks for all of your work and sharing!

**Comment by gab696** [comment]
for those having problems with nfs share not reconnecting on reboot, you can copy the entry created in /etc/mtab after the manual  mount to /etc/fstab

**Comment by LatestStop** [comment]
Thanks so much for taking the time!  Can't wait to try this out!
